dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/resourcetable/table/CategoryResourceTable.java

```diff
@@ -63,22 +65,27 @@ public ResourceTableType getTableType()
     @Override
     public String getCreateTempTableStatement()
     {
-        UniqueNameVerifier uniqueNameVerifier = new UniqueNameVerifier();
+        Set<String> usedColumnNames = new HashSet<>();
 
         String statement = "create table " + getTempTableName() + " (" +
             "categoryoptioncomboid bigint not null, " +
             "categoryoptioncomboname varchar(255), ";
 
         for ( Category category : objects )
         {
-            statement += uniqueNameVerifier.ensureUniqueShortName( category ) + " varchar(230), ";
+            statement += quote( category.getShortName() ) + " varchar(230), ";
             statement += quote( category.getUid() ) + " character(11), ";
+            usedColumnNames.add( category.getShortName() );
         }
 
         for ( CategoryOptionGroupSet groupSet : groupSets )
         {
-            statement += uniqueNameVerifier.ensureUniqueShortName( groupSet ) + " varchar(230), ";
+            String columnName = usedColumnNames.contains( groupSet.getShortName() )
+                ? "column" + usedColumnNames.size()
```

Comment:
Could we centralize this uniqify logic in a method or class? Will make it more expressive and testable.

Reply:
I just removed that because I felt we only need it here in this special situation and all it is, is a lookup in a set which I felt is not complex enough for it being a util on it own. But we could just resurrect the util we had.

Reply:
Right, I know this is a finer point. Just think the code will get more expressive if put in a properly named util method.

Reply:
I think we could improve the name in the case of a uniqueness constraint. Short name is max 50 chars and Postgres column name is max 63. So instead of "column[n]" we could do "groupSet.shortName[n]".

Reply:
@larshelge you mean `<groupSet.shortName><n>` as a pattern where we replace the name with the actual short name of the group set and add a number N for the nth column to make it unique?

Reply:
Right but just in the case where we have a uniqueness conflict. I.e. if the short name is not unique/collides with another short name of a group/category, then append a number to make it unique.

Reply:
I resurrected and adopted the existing unique name helper to create the pattern and added new tests for it.